### PR TITLE
Redirect easily changeable backend functions straight to core

### PR DESCRIPTION
**Pull request by ericdnielsen**
Like the previous PR this removes very simply forwarding functions in tfjs_backend.ts that simply forward to a similar function in -core.

Renames:
subtract->sub
multiply->mul
divide->div
clip->clipByValue

Removes internal usages of the permuteDimensions alias to transpose and just uses transpose

Changes layers code to use the core calling convention on squeeze (caller wrapper in the array) and removes our wrapper.

Functions that only varied in unused extra arguments:
zeros, ones, randomUniform, truncatedNormal, softmax

There's still a few more "easy" conversions but I think they'll need a touch more discussion for instance (scalarPlusArray, scalarTimesArray which are simple stricter type checks around add/mul  -- or replace layer's square(x) with core's mul(x,x)

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-layers/170)

### CODE REVIEW

**Comment by davidsoergel** [comment]
:lgtm_strong:

---

Reviewed 18 of 18 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-layers/170#-:-LC59U_Q7PcT4P7wBshu:b-jbsf8z)*

**Comment by caisq** [comment]
:lgtm_strong:
Thanks.

---

Review status: 16 of 18 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---

*[src/initializers.ts, line 374 at r2](https://reviewable.io/reviews/tensorflow/tfjs-layers/170#-LC5DWdd0cPcqxaRLg87:-LC5DWdd0cPcqxaRLg88:bgvlb14) ([raw file](https://github.com/tensorflow/tfjs-layers/blob/046e7e139bc499a01c364f6d8177d1a8654f8198/src/initializers.ts#L374)):*
> ```
> truncatedNormal
> ```

s/truncatedNormal/varianceScaling/ Maybe even use `this.getClassName()`, because this may be a subtype of VarianceScaling.

---


*Comments from [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-layers/170#-:-LC5DgwG4ua0utKEOTYR:bkjvnc8)*

**Comment by ericdnielsen** [comment]
Review status: 15 of 18 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[src/initializers.ts, line 374 at r2](https://reviewable.io/reviews/tensorflow/tfjs-layers/170#-LC5DWdd0cPcqxaRLg87:-LC5Eg-V2IBPuUVzcBVh:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-layers/blob/046e7e139bc499a01c364f6d8177d1a8654f8198/src/initializers.ts#L374)):*
<details><summary><i>Previously, caisq (Shanqing Cai) wrote…</i></summary><blockquote>

> ```
> truncatedNormal
> ```

s/truncatedNormal/varianceScaling/ Maybe even use `this.getClassName()`, because this may be a subtype of VarianceScaling.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-layers/170)*
